packages/flutter/lib/src/widgets/animated_list.dart

```diff
@@ -333,7 +332,32 @@ class AnimatedListState extends State<AnimatedList> with TickerProviderStateMixi
         ..sort();
     });
 
-    controller.reverse().then<void>((void value) {
+    if (builder != null && duration != Duration.zero) {
+      controller.reverse().then<void>((void value) {
+        _removeActiveItemAt(_outgoingItems, outgoingItem.itemIndex).controller.dispose();
+
+        // Decrement the incoming and outgoing item indices to account
+        // for the removal.
+        for (_ActiveItem item in _incomingItems) {
+          if (item.itemIndex > outgoingItem.itemIndex)
+            item.itemIndex -= 1;
+        }
+        for (_ActiveItem item in _outgoingItems) {
+          if (item.itemIndex > outgoingItem.itemIndex)
+            item.itemIndex -= 1;
+        }
+
+        setState(() {
+          _itemsCount -= 1;
+        });
+      });
+    }
+    /// In cases when the playback of an animation is not desired.
```

Comment:
Can you remove this comment? It doesn't really help in understanding the code. This should be in the PR description.